avocado/plugins/runner_nrunner.py

```diff
@@ -359,8 +359,8 @@ def run_suite(self, job, test_suite):
                     if rt.task.category == 'test']
         tsm = TaskStateMachine(self.runtime_tasks, self.status_repo)
         spawner_name = test_suite.config.get('nrunner.spawner')
-        spawner = SpawnerDispatcher(test_suite.config)[spawner_name].obj
-        spawner.job_output_dir = job.test_results_path
+        spawner = SpawnerDispatcher(test_suite.config,
+                                    job.test_results_path)[spawner_name].obj
```

Comment:
This is much better and indeed what I originally suggested. But I still wonder can't the `test_results_path` be integrated inside the test_suite config? I understand one could argue that the job configuration has greater scope than the test suite configuration but if the path is used inside a spawner inside a test suite, it seems legitimate to assume it is also test suite configuration, no?

Reply:
@pevogam do you mean having a different configuration for the name of the "test-results" component (currently hardcoded to `test-results`)?  That could be something that would allow different test suites to write to different directories, in contrast to the current situation where all test results of all test suites go to the same directory.

Or, do you simply mean not having an extra argument to `SpanwerDispatcher`?

Reply:
I see your point @pevogam. 

I think the idea so far, is to have on `config dicts` only configs sets by the user (`avocado config reference`), `job.test_results_dir` is something that is computed, based on `run.results_dir` and on the `job.unique_id`. Although I think it's valid to populate the dict with other "computed internal values", maybe it is just something that it was not much discussed at this point and needs a better planning, at least in terms of namespace, IMO. So I would suggest this move for the future.

In meanwhile maybe we just need to pass the `job` pointer object as proposed on my previous comment. So the spawner could get access to job.output_dir or `job.tests_results_path`.

But what maybe I'm missing here, are the situations where the spawner is called without the job property set. @richtja can you give an example please?

Reply:
Yes, I think all of this was resolved and improved in @richtja's most recent push.